template.tpl

```diff
@@ -60,59 +60,124 @@ ___TEMPLATE_PARAMETERS___
+        "simpleValueType": true,
+        "canBeEmptyString": true,
+        "valueHint": "E.g. https://cdn.liveshopping.bambuser.com/metrics/bambuser.min.js",
+        "valueValidators": [
+          {
+            "type": "REGEX",
+            "args": [
+              "(^$|^(https?:\\/\\/)?([\\da-z\\.-]+\\.[a-z\\.]{2,6}|[\\d\\.]+)([\\/:?\u003d\u0026#]{1}[\\da-z\\.-]+)*[\\/\\?]?$)"
+            ],
+            "errorMessage": "Please insert a valid URL including!",
+            "enablingConditions": []
+          }
+        ],
+        "help": "Do not write anything here unless you have been given a custom tracking library URL!"
+      }
+    ],
+    "groupStyle": "ZIPPY_CLOSED"
   }
 ]
 
 
 ___SANDBOXED_JS_FOR_WEB_TEMPLATE___
 
 // Enter your template code here.
-const callInWindow = require('callInWindow');
+const injectScript = require('injectScript');
+const queryPermission = require('queryPermission');
 const copyFromWindow = require('copyFromWindow');
-const getCookieValues = require('getCookieValues');
-const log = require('logToConsole');
-
-var bbu = copyFromWindow('_bambuser');
-if (bbu != null && typeof(bbu) != 'undefined') {
-  var eventData = {};
-  eventData[bbu.params.EVENT.dataSelector] = 'purchase';
-  eventData[bbu.params.ORDER_ID.dataSelector] = data.orderId;
-  eventData[bbu.params.ORDER_VALUE.dataSelector] = data.orderValue;
-  eventData[bbu.params.CURRENCY.dataSelector] = data.currency;
-  eventData[bbu.params.ORDER_PRODUCTIDS.dataSelector] = data.productIds;
-  
-  bbu.collect(eventData);
-}
-else
+const logToConsole = require('logToConsole');
+
+// If the user chose to log debug output, initialize the logging method
+const log = data.debug
+  ? logToConsole
+  : function () {};
+
+// Get the URL the user input into the text field
+const url = data.scriptUrl || 'https://cdn.liveshopping.bambuser.com/metrics/bambuser.min.js';
+
+log('🧪 Bambuser Debugger 🧪\n', '🏁 Bambuser purchase tracker started with the given tag data: ', data);
+
+// If the script loaded successfully, log a message and signal success
+const onSuccess = () => {
+  log('🧪 Bambuser Debugger 🧪\n', '✅ Tracking Library loaded successfully.');
+  var bbu = copyFromWindow('_bambuser');
+  if (bbu != null && typeof (bbu) != 'undefined') {
+    var eventData = {};
+    eventData[bbu.params.EVENT.dataSelector] = 'purchase';
+    eventData[bbu.params.ORDER_ID.dataSelector] = data.orderId;
+    eventData[bbu.params.ORDER_VALUE.dataSelector] = data.orderValue;
+    eventData[bbu.params.CURRENCY.dataSelector] = data.currency;
+    eventData[bbu.params.ORDER_PRODUCTIDS.dataSelector] = data.productIds;
+
+    bbu.collect(eventData);
+    log('🧪 Bambuser Debugger 🧪\n', '✅ Purchase has been tracked successfully with the following order data:', eventData);
+    // Call data.gtmOnSuccess when the tag is finished.
+    data.gtmOnSuccess();
+  }
+  else
+    log('🧪 Bambuser Debugger 🧪\n', '❌ Problem with loading the library: ', bbu);
+    data.gtmOnFailure();
+};
```

Comment:
It looks like you're missing a `}` somewhere around here.

Reply:
Good catch! 🦅
It should look cleaner now.